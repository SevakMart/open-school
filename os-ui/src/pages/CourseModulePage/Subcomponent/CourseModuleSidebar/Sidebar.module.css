.Sidebar_container {
  background: #ffffff;
  height: 992px;
  padding-left: 48px;
  transform-origin: top left;
}
.Sidebar_container .ArrowRightIcon {
  background: #ffffff;
  border: none;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  color: #333941;
  align-items: center;
}
.Sidebar_container .chevronIsOpen {
  background: #ffffff;
  cursor: pointer;
  transform: rotate(90deg);
  transition: 0.5s;
  margin-left: 15px;
}
.Sidebar_container .chevronIsClosed {
  background: #ffffff;
  cursor: pointer;
  transform: rotate(0deg);
  transition: 0.5s;
  margin-left: 15px;
}
.Sidebar_container .moduleDescriptionIsOpen {
  margin-left: 25px;
  width: 90%;
}
.Sidebar_container .moduleDescriptionIsClosed {
  display: none;
}
.Sidebar_container .Sidebar_courseName {
  color: #5E617B;
  font-weight: 600;
  font-size: 20px;
}
.Sidebar_container .SidebarOverview_button {
  margin-top: 32px;
  color: #333941;
  font-size: 16px;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
}
.Sidebar_container .SidebarOverview_button:active {
  color: #222427;
  font-weight: 700;
}
.Sidebar_container .Checkbox_container {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 36px;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;
  display: flex;
  align-items: center;
  color: #5E617B;
}
.Sidebar_container .Checkbox_container .Checkbox_checkmark {
  position: absolute;
  left: 0;
  height: 24px;
  width: 24px;
  background-color: #EEEFF4;
  border-radius: 15px;
}
.Sidebar_container .Checkbox_container .Checkbox_checkmark:after {
  content: "";
  position: absolute;
  display: none;
}
.Sidebar_container .Checkbox_container input {
  height: 0;
  width: 0;
}
.Sidebar_container .Checkbox_container input:checked ~ .Checkbox_checkmark {
  background-color: #4AB8A3;
}
.Sidebar_container .Checkbox_container input:checked ~ .Checkbox_checkmark:after {
  display: block;
}
.Sidebar_container .Checkbox_container .Checkbox_checkmark:after {
  left: 9px;
  top: 6px;
  width: 4px;
  height: 8px;
  border: solid #ffffff;
  border-width: 0 3px 3px 0;
  transform: rotate(45deg);
}/*# sourceMappingURL=Sidebar.module.css.map */