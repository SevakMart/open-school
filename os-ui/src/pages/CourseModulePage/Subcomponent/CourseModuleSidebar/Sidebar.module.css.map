{"version":3,"sources":["Sidebar.module.scss","../../../../styles/Configs/Colors.scss","Sidebar.module.css"],"names":[],"mappings":"AAEA;EACI,mBCCe;EDAf,aAAA;EACA,kBAAA;EACH,0BAAA;AEDD;AFGC;EACC,mBCLiB;EDMX,YAAA;EACA,yBAAA;EACN,kBAAA;EACA,gBAAA;EACA,eAAA;EAEM,cAAA;EAEN,mBAAA;AEHF;AFOC;EACC,mBCnBiB;EDoBX,eAAA;EACA,wBAAA;EACA,gBAAA;EACA,iBAAA;AELR;AFOI;EACF,mBC1BiB;ED2BX,eAAA;EACA,uBAAA;EACA,gBAAA;EACA,iBAAA;AELR;AFOI;EACI,iBAAA;EACA,UAAA;AELR;AFQI;EACI,aAAA;AENR;AFSI;EACE,cC5CY;ED6CZ,gBAAA;EACA,eAAA;AEPN;AFSI;EACE,gBAAA;EACA,cC/Ca;EDgDb,eAAA;EACA,uBAAA;EACA,YAAA;EACA,eAAA;EACA,UAAA;AEPN;AFSI;EACI,cCzDY;ED0DZ,gBAAA;AEPR;AFSI;EACE,cAAA;EACA,kBAAA;EACA,kBAAA;EACA,mBAAA;EACA,eAAA;EACA,gBAAA;EACA,eAAA;EACA,aAAA;EACA,mBAAA;EACA,cCvEY;ACgElB;AFSQ;EACE,kBAAA;EACA,OAAA;EACA,YAAA;EACA,WAAA;EACA,yBCzEU;ED0EV,mBAAA;AEPV;AFSQ;EACE,WAAA;EACA,kBAAA;EACA,aAAA;AEPV;AFUI;EACE,SAAA;EACA,QAAA;AERN;AFUI;EACE,yBCtFmB;AC8EzB;AFUI;EACE,cAAA;AERN;AFUI;EACE,SAAA;EACA,QAAA;EACA,UAAA;EACA,WAAA;EACA,qBAAA;EACA,yBAAA;EACA,wBAAA;AERN","file":"Sidebar.module.css"}