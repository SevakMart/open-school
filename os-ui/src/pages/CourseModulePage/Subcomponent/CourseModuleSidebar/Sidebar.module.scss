@import "../../../../styles/Configs/Colors.scss";

.Sidebar_container {
    background: $white-text-color;
    height: 992px;
    padding-left: 48px;
	  transform-origin: top left;
	.ArrowRightIcon{
		background: $white-text-color;
        border: none;
        font-family: 'Montserrat';
		font-style: normal;
		font-weight: 500;
		font-size: 16px;
		// line-height: 24px;
        color: #333941;
		// display: flex;
		align-items: center;
		// margin-bottom: 21,5px;
		// margin-top:24px;
	}
	.chevronIsOpen {
		background: $white-text-color;
        cursor:pointer;
        transform:rotate(90deg);
        transition: 0.5s;
        margin-left: 15px;
    }
    .chevronIsClosed {
		background: $white-text-color;
        cursor:pointer;
        transform:rotate(0deg);
        transition: 0.5s;
        margin-left: 15px;
    }
    .moduleDescriptionIsOpen{
        margin-left: 25px;
        width: 90%;
        // @include fontStyle($primary-font,normal,400,15px,70px);
    }
    .moduleDescriptionIsClosed {
        display:none;
    }

    .Sidebar_courseName {
      color: $main-text-color;
      font-weight: 600;
      font-size: 20px;
    }
    .SidebarOverview_button {
      margin-top: 32px;
      color: $black-text-color;
      font-size: 16px;
      background: transparent;
      border: none;
      cursor: pointer;
      padding: 0;
    }
    .SidebarOverview_button:active {
        color: $header-main-color;
        font-weight: 700;
    }
    .Checkbox_container {
      display: block;
      position: relative;
      padding-left: 35px;
      margin-bottom: 36px;
      cursor: pointer;
      font-weight: 400;
      font-size: 14px;
      display: flex;
      align-items: center;
      color: $main-text-color;

        .Checkbox_checkmark {
          position: absolute;
          left: 0;
          height: 24px;
          width: 24px;
          background-color: $main-course-color;
          border-radius: 15px;
        }
        .Checkbox_checkmark:after {
          content: "";
          position: absolute;
          display: none;
        }
    }
    .Checkbox_container input {
      height: 0;
      width: 0;
    }
    .Checkbox_container input:checked ~ .Checkbox_checkmark {
      background-color: $checkbox-checked-color;
    }
    .Checkbox_container input:checked ~ .Checkbox_checkmark:after {
      display: block;
    }
    .Checkbox_container .Checkbox_checkmark:after {
      left: 9px;
      top: 6px;
      width: 4px;
      height: 8px;
      border: solid $white-text-color;
      border-width: 0 3px 3px 0;
      transform: rotate(45deg);
    }
    
  .btnClickedCName {
    text-decoration: none; 
    color: #415791;
    font-weight: 600;
    font-size: 20px;
    line-height: 24px;
  }
  .btnNotClickedCName {
    text-decoration: none; 
    color: #333941;
    font-weight: 500;
    font-size: 16px;
    line-height: 24px;
  }
}
