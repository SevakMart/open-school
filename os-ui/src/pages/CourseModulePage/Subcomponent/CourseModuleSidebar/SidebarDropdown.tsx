import { t } from 'i18next';
import { COURSE_CATEGORIES } from '../../../../constants/CourseModuleCategories';
import styles from './Sidebar.module.scss';

interface Props {
handleChangeValue: (a:string) => void,
  }

const SidebarDropdown = ({ handleChangeValue }: Props) => (
  <>
    {
		COURSE_CATEGORIES.map((button: { value: string, id: string, title: string }) => (
  <label key={button.id} className={styles.Checkbox_container}>
    {t(`${button.title}`)}
    <input type="checkbox" onChange={() => handleChangeValue(button.value)} />
    <span className={styles.Checkbox_checkmark} />
  </label>
		))
	  }
  </>
);

export default SidebarDropdown;
