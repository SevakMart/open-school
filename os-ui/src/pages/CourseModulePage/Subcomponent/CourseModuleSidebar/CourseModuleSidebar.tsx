/* eslint-disable react/jsx-key */
/* eslint-disable react/button-has-type */
import { t } from 'i18next';
import Dropdown from '../../../../component/Dropdown/Dropdown';
import SidebarDropdown from './SidebarDropdown';
import styles from './Sidebar.module.scss';
import { COURSE_NAME } from '../../../../constants/CourseModuleCategories';

interface SidebarDropdownProps {
  value: string,
  handleChangeValue: (a:string) => void
}

const CourseModuleSidebar = ({ value, handleChangeValue }: SidebarDropdownProps) => (
  <div className={styles.Sidebar_container}>
    <h4 className={styles.Sidebar_courseName}>{t('Course Name')}</h4>
    <Dropdown
      open={value}
      trigger={<button className={styles.SidebarOverview_button}>{t('Overview')}</button>}
      menu={[
        <SidebarDropdown handleChangeValue={handleChangeValue} />,
      ]}
    />
    {
        COURSE_NAME.map((button) => (
          <div
            className={styles.SidebarOverview_button}
            key={button.id}
          >
            {button.desc}
          </div>
        ))
      }
  </div>
);
export default CourseModuleSidebar;
