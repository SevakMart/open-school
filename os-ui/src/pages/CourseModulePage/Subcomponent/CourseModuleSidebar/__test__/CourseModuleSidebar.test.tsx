import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from '@reduxjs/toolkit';
import { BrowserRouter } from 'react-router-dom';
import rootReducer from '../../../../../redux/RootReducer';
import CourseModuleSidebar from '../CourseModuleSidebar';

describe('CourseModuleSidebar', () => {
  // const handleChangeValueMock = jest.fn();
  // const modulesMock = [{ title: 'Module 1', description: 'Module 1 description', moduleItemSet: [{}, {}] },
  //   {
  //     title: 'Module 2',
  //     description: 'Module 2 description',
  //     moduleItemSet: [{}, {}],
  //   },
  // ];

  const store = createStore(rootReducer);

  test('renders the component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CourseModuleSidebar />
        </BrowserRouter>
      </Provider>,
    );
    // expect(screen.getByText('Course Name')).toBeInTheDocument();
    expect(screen.getByText('Overview')).toBeInTheDocument();
    // expect(screen.getByAltText('chevron')).toBeInTheDocument();
    // expect(screen.getByTestId('moduleListOpen')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('toggles the module list on click', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CourseModuleSidebar />
        </BrowserRouter>
      </Provider>,
    );
    fireEvent.click(screen.getByAltText('chevron'));
    // expect(screen.getByTestId('moduleListClosed')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(screen.getByAltText('chevron'));
    // expect(screen.getByTestId('moduleListOpen')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls handleChangeValue when clicking the Overview button', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CourseModuleSidebar />
        </BrowserRouter>
      </Provider>,
    );
    fireEvent.click(screen.getByText('Overview'));
    // expect(handleChangeValueMock).toHaveBeenCalledWith('Overview');
    expect(asFragment()).toMatchSnapshot();
  });
});
