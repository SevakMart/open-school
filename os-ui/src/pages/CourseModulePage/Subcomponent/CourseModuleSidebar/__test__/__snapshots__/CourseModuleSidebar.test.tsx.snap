// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseModuleSidebar calls handleChangeValue when clicking the Overview button 1`] = `
<DocumentFragment>
  <div
    class="Sidebar_container"
  >
    <h4
      class="Sidebar_courseName"
    />
    <div
      class="dropdown"
    >
      <button
        class="ArrowRightIcon"
        type="button"
      >
        <a
          class="btnClickedCName"
          href="/userCourse/modulOverview/undefined"
        >
          Overview
        </a>
        <img
          alt="chevron"
          class="chevronIsOpen"
          src="ArrowRight.svg"
        />
      </button>
    </div>
    <div
      class="SidebarOverview_button"
    >
      Additional Materials
    </div>
    <div
      class="SidebarOverview_button"
    >
      Grade
    </div>
    <div
      class="SidebarOverview_button"
    >
      <a
        class="btnNotClickedCName"
        href="/userCourse/modulOverview/undefined/discussionForum"
      >
        Discussion Forum
      </a>
    </div>
    <div
      class="SidebarOverview_button"
    >
      FAQ
    </div>
  </div>
</DocumentFragment>
`;

exports[`CourseModuleSidebar renders the component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar_container"
  >
    <h4
      class="Sidebar_courseName"
    />
    <div
      class="dropdown"
    >
      <button
        class="ArrowRightIcon"
        type="button"
      >
        <a
          class="btnClickedCName"
          href="/userCourse/modulOverview/undefined"
        >
          Overview
        </a>
        <img
          alt="chevron"
          class="chevronIsOpen"
          src="ArrowRight.svg"
        />
      </button>
    </div>
    <div
      class="SidebarOverview_button"
    >
      Additional Materials
    </div>
    <div
      class="SidebarOverview_button"
    >
      Grade
    </div>
    <div
      class="SidebarOverview_button"
    >
      <a
        class="btnNotClickedCName"
        href="/userCourse/modulOverview/undefined/discussionForum"
      >
        Discussion Forum
      </a>
    </div>
    <div
      class="SidebarOverview_button"
    >
      FAQ
    </div>
  </div>
</DocumentFragment>
`;

exports[`CourseModuleSidebar toggles the module list on click 1`] = `
<DocumentFragment>
  <div
    class="Sidebar_container"
  >
    <h4
      class="Sidebar_courseName"
    />
    <div
      class="dropdown"
    >
      <button
        class="ArrowRightIcon"
        type="button"
      >
        <a
          class="btnClickedCName"
          href="/userCourse/modulOverview/undefined"
        >
          Overview
        </a>
        <img
          alt="chevron"
          class="chevronIsClosed"
          src="ArrowRight.svg"
        />
      </button>
    </div>
    <div
      class="SidebarOverview_button"
    >
      Additional Materials
    </div>
    <div
      class="SidebarOverview_button"
    >
      Grade
    </div>
    <div
      class="SidebarOverview_button"
    >
      <a
        class="btnNotClickedCName"
        href="/userCourse/modulOverview/undefined/discussionForum"
      >
        Discussion Forum
      </a>
    </div>
    <div
      class="SidebarOverview_button"
    >
      FAQ
    </div>
  </div>
</DocumentFragment>
`;

exports[`CourseModuleSidebar toggles the module list on click 2`] = `
<DocumentFragment>
  <div
    class="Sidebar_container"
  >
    <h4
      class="Sidebar_courseName"
    />
    <div
      class="dropdown"
    >
      <button
        class="ArrowRightIcon"
        type="button"
      >
        <a
          class="btnClickedCName"
          href="/userCourse/modulOverview/undefined"
        >
          Overview
        </a>
        <img
          alt="chevron"
          class="chevronIsOpen"
          src="ArrowRight.svg"
        />
      </button>
    </div>
    <div
      class="SidebarOverview_button"
    >
      Additional Materials
    </div>
    <div
      class="SidebarOverview_button"
    >
      Grade
    </div>
    <div
      class="SidebarOverview_button"
    >
      <a
        class="btnNotClickedCName"
        href="/userCourse/modulOverview/undefined/discussionForum"
      >
        Discussion Forum
      </a>
    </div>
    <div
      class="SidebarOverview_button"
    >
      FAQ
    </div>
  </div>
</DocumentFragment>
`;
